### PR TITLE
Release CumulusCI 3.52.0

### PR DESCRIPTION
**Pull request by davidjray**

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -2,7 +2,26 @@
 History
 =======
 
+3.52.0 (2022-02-03)
+-------------------
+
+Critical Changes
+
+* We fixed a bug where the ``Run Task`` and ``Run Task Class`` robot keywords could throw an error like ``AttributeError: module 'robot.api.logger' has no attribute 'log'`` in rare cases (#3053)
```
Comment (davisagli) [inline HISTORY.rst:10]: This belongs under Issued Closed, not Critical Changes (critical changes is for when we changed something in a backwards incompatible way)

File: HISTORY.rst [hunk 2]
```diff
@@ -2,7 +2,26 @@
...
+
+Changes
+
+* Whenever possible, libdoc will generate relative pathnames when creating csv output (#3058)
```
Comment (davisagli) [inline HISTORY.rst:14]: context is not clear; this should mention the robot_libdoc task.

Comment (pconrad-sfdo) [inline HISTORY.rst:14]:
```suggestion
* Whenever possible, libdoc will generate relative pathnames when creating CSV output (#3058)
```

File: HISTORY.rst (hunk 1 above)
Comment (pconrad-sfdo) [inline HISTORY.rst:10]:
```suggestion
* We fixed a bug where the ``Run Task`` and ``Run Task Class`` robot keywords throw an error like ``AttributeError: module 'robot.api.logger' has no attribute 'log'`` in rare cases (#3053)
```

File: HISTORY.rst [hunk 3]
```diff
@@ -2,7 +2,26 @@
...
+* Added a new option ``--preview`` to the ``robot_libdoc`` task. When set to true it will automatically open up a browser window to the generated documentation. (#3057)
+* The robot keyword ``Locate element by label`` has been removed from the Salesforce.py library. This wasn't designed to be a keyword but was accidentally exported as one. If you want to find an input or textarea element by its label you can use a locator of the form ``label:<text>`` (eg: ``label:First Name``) (#3048)
+
+Issues Closed
+
+* Fixed an issue where overriding flow steps was not working as documented. (#3043)
```
Comment (davisagli) [inline HISTORY.rst:20]: @Br4nd0R should there also be something for this under Changes about the new simpler way to override a task with a flow and vice versa?

File: HISTORY.rst [hunk 4]
```diff
@@ -2,7 +2,26 @@
...
+-------------------
+
+Critical Changes
+
+* We fixed a bug where the ``Run Task`` and ``Run Task Class`` robot keywords could throw an error like ``AttributeError: module 'robot.api.logger' has no attribute 'log'`` in rare cases (#3053)
+
+Changes
+
+* Whenever possible, libdoc will generate relative pathnames when creating csv output (#3058)
+* Added a new option ``--preview`` to the ``robot_libdoc`` task. When set to true it will automatically open up a browser window to the generated documentation. (#3057)
```
Comment (pconrad-sfdo) [inline HISTORY.rst:15]:
```suggestion
* Added a new option ``--preview`` to the ``robot_libdoc`` task. When set to ``true`` it automatically opens a browser window to the generated documentation. (#3057)
```

**Review by davisagli — CHANGES_REQUESTED** [review]
This needs an update to version.txt, which is what will actually trigger the release once merged.

**Comment by davisagli** [comment]
We should probably also include release notes for:
- new version of snowfakery
- https://github.com/SFDO-Tooling/CumulusCI/pull/3028
- https://github.com/SFDO-Tooling/CumulusCI/pull/3059

**Comment by davidjray** [comment]
> This needs an update to version.txt, which is what will actually trigger the release once merged.

🤦🏽 

Good catch.

File: HISTORY.rst [hunk 5]
```diff
@@ -2,7 +2,23 @@
...
 =======
 
+3.52.0 (2022-02-03)
+-------------------
+
+Changes
+
+* Whenever possible, while running the ``robot_libdoc`` task libdoc generates relative pathnames when creating CSV output (#3058)
+* Added a new option ``--preview`` to the ``robot_libdoc`` task. When set to ``true`` it automatically opens a browser window to the generated documentation. (#3057)
+* The robot keyword ``Locate element by label`` has been removed from the Salesforce.py library. This wasn't designed to be a keyword but was accidentally exported as one. If you want to find an input or textarea element by its label you can use a locator of the form ``label:<text>`` (eg: ``label:First Name``) (#3048)
```
Comment (davisagli) [inline HISTORY.rst:12]:
```suggestion
* The robot keyword ``Locate element by label`` has been removed from the Salesforce.py library. This wasn't designed to be a keyword but was accidentally exported as one. If you want to find an input or textarea element by its label you can use a locator of the form ``label:<text>`` (eg: ``label:First Name``) (#3048)
* Updated to `Snowfakery 2.5.0 <https://github.com/SFDO-Tooling/Snowfakery/releases/tag/v2.5.0>`__
* If you have Python code that is importing CumulusCI's config classes, some of them have been reorganized into modules with a snake_case name. The old CamelCase imports should still work, but the new names are preferred for consistency with other parts of the codebase.
```

File: HISTORY.rst [hunk 6]
```diff
@@ -2,7 +2,23 @@
 History
 =======
 
+3.52.0 (2022-02-03)
+-------------------
+
+Changes
+
+* Whenever possible, while running the ``robot_libdoc`` task libdoc generates relative pathnames when creating CSV output (#3058)
```
Comment (davisagli) [inline HISTORY.rst:10]:
```suggestion
* Flow steps can now be replaced with the same syntax for all step types! (Current step types are: 'task' or 'flow'). See the [replacing a flow step](https://cumulusci.readthedocs.io/en/latest/config.html?highlight=override#replace-a-flow-step) docs for more details. CumulusCI is still compatible with the old syntax which required setting the current step type to ``None`` when replacing with a step of a differing type. (#3043)
* Whenever possible, while running the ``robot_libdoc`` task libdoc generates relative pathnames when creating CSV output (#3058)
```

File: HISTORY.rst [hunk 7]
```diff
@@ -2,7 +2,23 @@
...
+* Added a new option ``--preview`` to the ``robot_libdoc`` task. When set to ``true`` it automatically opens a browser window to the generated documentation. (#3057)
+* The robot keyword ``Locate element by label`` has been removed from the Salesforce.py library. This wasn't designed to be a keyword but was accidentally exported as one. If you want to find an input or textarea element by its label you can use a locator of the form ``label:<text>`` (eg: ``label:First Name``) (#3048)
+
+Issues Closed
+
+* Fixed an issue where overriding flow steps was not working as documented. (#3043)
+* We fixed a bug where the ``Run Task`` and ``Run Task Class`` robot keywords throw an error like ``AttributeError: module 'robot.api.logger' has no attribute 'log'`` in rare cases (#3053)
```
Comment (davisagli) [inline HISTORY.rst:17]:
```suggestion
* We fixed a bug where the ``Run Task`` and ``Run Task Class`` robot keywords throw an error like ``AttributeError: module 'robot.api.logger' has no attribute 'log'`` in rare cases (#3053)
* Fixed a bug in the `snowfakery` task which caused Unique IDs to not always be actually unique. (#3059)
```

**Review by davisagli — APPROVED** [review]

**Pull request merged**